p2p/discovery.py

```diff
@@ -445,6 +595,37 @@ def _unpack(message: bytes) -> Tuple[datatypes.PublicKey, int, List[Any], bytes]
     return remote_pubkey, cmd_id, payload, message_hash
 
 
+def _get_msg_expiration() -> int:
+    return int(time.time() + EXPIRATION)
+
+
+def _pack_v5(cmd_id: int, payload: List[Any], privkey: datatypes.PrivateKey) -> bytes:
+    """Create and sign a discovery v5 UDP message to be sent to a remote node."""
+    cmd_id = to_bytes(cmd_id)
+    encoded_data = cmd_id + rlp.encode(payload)
+    signature = privkey.sign_msg(encoded_data)
+    return signature.to_bytes() + encoded_data
+
+
+def _unpack_v5(message: bytes) -> Tuple[datatypes.PublicKey, int, List[Any], Hash32]:
```

Comment:
Side note.  We should document the following terminology

- message: the raw bytes that are sent over the wire between peers.
- payload: the data embedded in the message
- command: the abstraction for a message which includes decoding/encoding of payloads.

In another issue @carver and I talked about trying to normalize our naming since I think we currently use message in place of payload in many places.